src/cdk-experimental/listbox/listbox.ts

```diff
@@ -194,360 +158,382 @@ export class CdkOption<T = unknown> implements ListKeyManagerOption, Highlightab
+   * @param event The focusout event
    */
-  registerOnChange(fn: (value: T) => void): void {
-    this._onChange = fn;
+  protected _handleFocusOut(event: FocusEvent) {
+    const hostElement = this.elementRef.nativeElement;
+    const otherElement = event.relatedTarget as Element;
+    if (hostElement !== otherElement && !hostElement.contains(otherElement)) {
+      this._onTouched();
+    }
   }
 
-  /**
-   * Saves a callback function to be invoked when the select is blurred
-   * by the user. Required to implement ControlValueAccessor.
-   */
-  registerOnTouched(fn: () => {}): void {
-    this._onTouched = fn;
+  /** Get the id of the active option if active descendant is being used. */
+  protected _getAriaActiveDescendant(): string | null | undefined {
+    return this._useActiveDescendant ? this.listKeyManager?.activeItem?.id : null;
   }
 
-  /** Sets the select's value. Required to implement ControlValueAccessor. */
-  writeValue(values: T | T[]): void {
-    if (this._options) {
-      this._setSelectionByValue(values);
+  /** Initialize the key manager. */
+  private _initKeyManager() {
+    this.listKeyManager = new ActiveDescendantKeyManager(this.options)
+      .withWrap()
+      .withTypeAhead()
+      .withHomeAndEnd()
+      .withAllowedModifierKeys(['shiftKey']);
+
+    if (this.orientation === 'vertical') {
+      this.listKeyManager.withVerticalOrientation();
+    } else {
+      this.listKeyManager.withHorizontalOrientation(this._dir?.value || 'ltr');
     }
-  }
 
-  /** Disables the select. Required to implement ControlValueAccessor. */
-  setDisabledState(isDisabled: boolean): void {
-    this.disabled = isDisabled;
+    this.listKeyManager.change
+      .pipe(takeUntil(this.destroyed))
+      .subscribe(() => this._focusActiveOption());
   }
 
-  /** Returns the values of the currently selected options. */
-  getSelectedValues(): T[] {
-    return this._options.filter(option => option.selected).map(option => option.value);
+  // TODO(mmalerba): Should not depend on combobox.
+  private _updatePanelForSelection(option: CdkOption<T>) {
+    if (this._combobox) {
+      if (!this.multiple) {
+        this._combobox.updateAndClose(option.isSelected() ? option.value : []);
+      } else {
+        this._combobox.updateAndClose(this.value);
+      }
+    }
   }
 
-  /** Selects an option that has the corresponding given value. */
-  private _setSelectionByValue(values: T | T[]) {
-    for (const option of this._options.toArray()) {
-      this.deselect(option);
+  /** Update the selection mode when the 'multiple' property changes. */
+  private _updateSelectionMode() {
+    if (this.multiple !== this.selectionModel().isMultipleSelection()) {
+      let newSelection = this.value;
+      // If we're changing to a single selection model and there are multiple items selected,
+      // clear the selection rather than arbitrarily keeping one of the items selected.
+      // TODO(mmalerba): Is this the right behavior?
```

Comment:
This seems a little odd to me. I think it would make more sense to leave an invalid value and wait until the user interacts to change it to something else (similar to how text inputs can have invalid values)

Reply:
What if we just throw an error if you try to change the selection mode after initialization?

Reply:
I don't love that because it kind of violates what you expect from an `@Input()`. Now that we have a well defined order, just doing `.slice(0, 1)` on the current selection isn't sounding so bar. I checked `<select multiple>` and that's actually what it does

Reply:
Ok, after some messing around I've come back to treating it as nothing selected being the right thing to do (though this is different than what `<select multiple>` does.

Originally this code was updating the FormControl which is why it felt so wrong. Rather I think the right thing to do is don' change the form control value - just leave it as an invalid value and then add some Forms validator support so people can color the listbox red or whatever. But then as far as what we visually display for the invalid value, nothing selected makes the most sense because this allows the user to click any of the options to fix the value in the `FormControl`